social_core/pipeline/social_auth.py

```diff
@@ -5,6 +5,21 @@ def social_details(backend, details, response, *args, **kwargs):
     return {'details': dict(backend.get_user_details(response), **details)}
 
 
+def social_names(backend, details, response, *args, **kwargs):
+    # If first+last are both missing, populate from full
+    if details.get('fullname') and backend.setting('FIRSTLAST_FROM_FULL', True):
+        if not (details.get('first_name') or details.get('last_name')):
+            first, _space, last = details['fullname'].rpartition(' ')
+            details['first_name'] = first.strip()
+            details['last_name'] = last.strip()
+            print(f"end social_names {details=}")
```

Comment:
(I'll need to remove this, but I'd like to confirm that the general direction seems good before I do the cleanup.)